### PR TITLE
feat(optimus): [RND-164806] Add OptimusDivider

### PR DESCRIPTION
**Pull request by witwash**
#### Summary

- Added the `OptimusDivider`
- Added story for the `OptimusDivider`
<details><summary>Screenshots</summary>

</details>

#### Testing steps

1. Open Structure/Divider story.
2. Fiddle with the settings and check if the divider is displayed correctly.

#### Follow-up issues

None.

#### Check during review

- Verify against YouTrack issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).

### CODE REVIEW

File: optimus/lib/src/divider.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    Key? key,
+    this.text,
+    this.direction = Axis.horizontal,
+    this.variant = OptimusDividerVariant.normal,
+    this.crossAxisSpacing = spacing100,
+  }) : super(key: key);
+
+  /// The text to display in the middle of the divider. If null, the divider
+  /// will be just a line.
+  final Widget? text;
```
Comment (ookami-kb) [inline optimus/lib/src/divider.dart:17]: Since it's a widget, probably, it's better to go with `child`?

**Comment by witwash** [comment]
@ookami-kb, 
done

**Comment by witwash** [comment]
@ookami-kb,
any idea what is happening? 
CleanShot 2023-03-28 at 12 33 34

**Comment by witwash** [comment]
nvm, found it. Thought it look similar.

**Comment by ookami-kb** [comment]
@witwash yep, check this one: https://github.com/MewsSystems/tech-mo/pull/1853

**Comment by witwash** [comment]
@ookami-kb,
fixed

**Comment by ookami-kb** [comment]
@witwash please make it a separate PR, these change have nothing to do with the scope of this PR.

**Comment by witwash** [comment]
@ookami-kb,
ping

**Review by ookami-kb — APPROVED** [review]

**Pull request merged**